### PR TITLE
KEP-1847 v1.32 stable docs changesAutodelete 132

### PR DESCRIPTION
**Pull request by mattcary**
### Description
Docs changes for [KEP-1847](https://github.com/kubernetes/enhancements/issues/1847) going stable.

### CODE REVIEW

**Review by dipesh-rawat — COMMENTED** [review]
@mattcary, we should consider setting the **_dev-1.32_** branch as the target branch rather than the main branch for this pull request. Since the change is related to the feature targeting stable in 1.32.

**Comment by mattcary** [comment]
> @mattcary, we should consider setting the **_dev-1.32_** branch as the target branch rather than the main branch for this pull request. Since the change is related to the feature targeting stable in 1.32.

Oops, I meant to do that. Thx.

**Comment by tengqm** [comment]
The change to the api-ref-generator submodule seems unintended.

**Comment by mattcary** [comment]
> The change to the api-ref-generator submodule seems unintended.

Hmm, right you are. Submodules seem weird, I'm not able to revert the change

I tried `git checkout upstream/dev-1.32 -- api-ref-generator`, but that didn't change anything. Do you have any other suggestions?

**Comment by tengqm** [comment]
> > The change to the api-ref-generator submodule seems unintended.
> 
> Hmm, right you are. Submodules seem weird, I'm not able to revert the change
> 
> I tried `git checkout upstream/dev-1.32 -- api-ref-generator`, but that didn't change anything. Do you have any other suggestions?

Well ... my experience on submodules is pretty limited. Usually I'd do a clean commit again.

**Comment by mattcary** [comment]
Hurrah! I had to do a git submodules update from the dev-1.32 branch and then commit those changes. Anyway it seems clean now.

File: content/en/docs/reference/command-line-tools-reference/feature-gates/stateful-set-auto-delete-pvc.md [hunk 1]
```diff
@@ -14,9 +14,12 @@ stages:
   - stage: beta
     defaultValue: true
     fromVersion: "1.27"
```
Comment (tengqm) [inline content/en/docs/reference/command-line-tools-reference/feature-gates/stateful-set-auto-delete-pvc.md:16]: toVersion: "1.31"

Comment (mattcary) [reply content/en/docs/reference/command-line-tools-reference/feature-gates/stateful-set-auto-delete-pvc.md:16]: done

File: content/en/docs/concepts/workloads/controllers/statefulset.md [hunk 2]
```diff
@@ -386,7 +386,7 @@ StatefulSet will then begin to recreate the Pods using the reverted template.
 
 ## PersistentVolumeClaim retention
 
-{{< feature-state for_k8s_version="v1.27" state="beta" >}}
+{{< feature-state for_k8s_version="v1.31" state="stable" >}}
```
Comment (tengqm) [inline content/en/docs/concepts/workloads/controllers/statefulset.md:389]: `{{< feature-state feature_state_name="StatefuleSetAutoDeletePVC" >}}`

Comment (mattcary) [reply content/en/docs/concepts/workloads/controllers/statefulset.md:390]: Done. Further up in the document is the tag feature_gate_name, is that what I should use?

Comment (tengqm) [reply content/en/docs/concepts/workloads/controllers/statefulset.md:390]: I think so. The shortcode can/should be removed when the gate is removed.

Comment (tengqm) [reply content/en/docs/concepts/workloads/controllers/statefulset.md:390]: Oh ..., you may want to remove line 390.

Comment (mattcary) [reply content/en/docs/concepts/workloads/controllers/statefulset.md]: done

**Comment by tengqm** [comment]
The upstream change (https://github.com/kubernetes/kubernetes/pull/116501) has been merged. So
/approve

**Comment by mattcary** [comment]
> The upstream change ([kubernetes/kubernetes#116501](https://github.com/kubernetes/kubernetes/pull/116501)) has been merged. So /approve

Oops, [#128247](https://github.com/kubernetes/kuberenetes/pull/128247) is the PR for feature gate to stable.

**Comment by tengqm** [comment]
Looks like the PR I found was not related to this change.
/hold

@mattcary The PR from your comment points to 404. Please check. Thanks.

**Comment by mattcary** [comment]
Sorry, this is the link: https://github.com/kubernetes/kubernetes/pull/128247

**Comment by mattcary** [comment]
@tengqm : https://github.com/kubernetes/kubernetes/pull/128247 has been merged, thanks for your patience.

**Pull request merged**